homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -411,12 +503,38 @@ def __init__(
         """Initialize Envoy entity."""
         super().__init__(coordinator, description)
         self._attr_unique_id = f"{self.envoy_serial_num}_{description.key}"
+        ct_count = coordinator.envoy.ct_meter_count
+        ct_consumptionmeter = coordinator.envoy.consumption_meter_type
+        phase_count = coordinator.envoy.phase_count
+        # Display phase count and found CT meters in model
+        model = (
+            "Envoy "
+            # Show CT Phase mode if CT is found
+            + (
+                f", phase mode: {coordinator.envoy.phase_mode}"
+                if ct_consumptionmeter is not None
+                else ""
+            )
+            # show number of phases found in Model description
+            + f" , phases: {str(phase_count)}"
+            # Show production CT found if 1 CT and no consumption ct of if more CT found
+            + (
+                ", production CT"
+                if (ct_count == 1 and ct_consumptionmeter is None) or (ct_count > 1)
+                else ""
+            )
+            # Add type of consumption CT to model if one was found
+            + (f", {ct_consumptionmeter} CT" if ct_consumptionmeter is not None else "")
+            + (", no CT" if ct_count == 0 else "")
+        )
```

Comment:
I have mixed feelings about this part. It's very unclear what is happening when. I am leaning towards extracting this into a function, but that would require a lot of parameters. Or you would have to pass in the coordinator. 

(The biggest factor that makes this worse are the inline ternaries)

Reply:
Intent is to show phase features with the model. Maybe this makes it more clear:
```Py
    model = f"Envoy, phases: {phase_count}"
    if ct_consumptionmeter:
        # Add CT phase mode to model if consumption CT is found
        model = f"{model}, phase mode: {coordinator.envoy.phase_mode}"
        # Add type of consumption CT to model if CT is found
        model = f"{model}, {ct_consumptionmeter} CT"
    if (ct_count == 1 and ct_consumptionmeter is None) or (ct_count > 1)
        # Add Production CT to model if present
        # if only 1 ct and no consumption ct found or more as 1 ct found
        model = f"{model}, production CT"
    if ct_count == 0:
        model = f"{model}, no CT"
```

Reply:
I think it looks better, but I do think we should put it in a separate function

Reply:
moved to seperate function